notebooks/HASP/DataDiagnostic/DataDiagnostics.ipynb

```diff
@@ -0,0 +1,1599 @@
+{
```

Comment:
<p>When the log is large, reading it all at once with <code>f.readlines()</code> is not very memory efficient. Instead, consider iterating over the file object directly. This approach reads one line at a time and is more memory-efficient for large files: </p><p> </p><pre class="ql-syntax" spellcheck="false">with open(logfile, 'r') as f:
  for line in f:
    print(line.strip()) 
</pre>
<br/>

 _Reply via <a href='https://app.reviewnb.com/spacetelescope/hst_notebooks/pull/207/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='21'/>